src/lightning_habana/pytorch/strategies/fsdp.py

```diff
@@ -0,0 +1,254 @@
+            auto_wrap_policy=auto_wrap_policy,
+            activation_checkpointing=activation_checkpointing,
+            activation_checkpointing_policy=activation_checkpointing_policy,
+            sharding_strategy=sharding_strategy,
+            state_dict_type=state_dict_type,
+        )
+
+    @property
+    def mixed_precision_config(self) -> Optional["MixedPrecision"]:
+        if self.mixed_precision:
+            return self.mixed_precision
+        plugin = self.precision_plugin
+        if isinstance(plugin, HPUFSDPPrecision):
+            return plugin.mixed_precision_config
+        return None
+
+    @property  # type: ignore[override]
+    @override
+    def precision_plugin(self) -> HPUFSDPPrecision:
+        plugin = self._precision_plugin
+        if plugin is not None:
+            return plugin
+        return HPUFSDPPrecision("bf16-mixed")
+
+    @precision_plugin.setter
+    @override
+    def precision_plugin(self, precision_plugin: Optional[HPUFSDPPrecision]) -> None:
+        if precision_plugin is not None and not isinstance(precision_plugin, HPUFSDPPrecision):
+            raise TypeError(
+                f"The FSDP strategy can only work with the `HPUFSDPPrecision` plugin, found {precision_plugin}"
+            )
+        self._precision_plugin = precision_plugin
+
+    @override
+    def setup_environment(self) -> None:
+        if self._process_group_backend == "hccl":
+            # this env is used in overrides to check the backend initiated
+            _ws = self.cluster_environment.world_size()
+            _grank = self.cluster_environment.global_rank()
+            _lrank = self.cluster_environment.local_rank()
+            hpu_dist.initialize_distributed_hpu(world_size=_ws, rank=_grank, local_rank=_lrank)
+        super().setup_environment()
+
+    @override
+    def _setup_model(self, model: Module) -> Module:
+
+        from torch.distributed.fsdp import FullyShardedDataParallel
+
+        if any(isinstance(mod, FullyShardedDataParallel) for mod in model.modules()):
+            if _has_meta_device_parameters(model):
+                rank_zero_warn(
+                    "The model is already wrapped in `FSDP` but there are still parameters on the meta device."
+                )
+            if "auto_wrap_policy" in self.kwargs:
+                # The user has wrapped their submodules manually, don't apply the auto wrap policy.
+                rank_zero_warn(
+                    "A FSDP `auto_wrap_policy` is set, but the model is already wrapped. The policy will be ignored."
+                )
+                del self.kwargs["auto_wrap_policy"]
+        else:
+            model = FullyShardedDataParallel(
+                module=model,
+                cpu_offload=self.cpu_offload,
+                mixed_precision=self.mixed_precision_config,
+                sharding_strategy=self.sharding_strategy,
+                device_id=self.root_device,  # Index based device selection is not supported on HPU
```

Comment:
This change could be done in Lightning directly, and then you could avoid the override.

Reply:
Agreed. The thought process was to avoid changes to lightning core and absorb changes in the plugins so that both repositories needn't be made dependent for a release. 
Do you want us to push a change to lightning fsdp strategy ?

Reply:
We'll relook at this cleanup once lightning base changes to accommodate all accelerators.